File: app/src/main/java/fr/narwhals/go/domain/Point.java
package fr.narwhals.go.domain;

import java.io.Serializable;

public class Point implements Serializable {
	private static final long serialVersionUID = -2116601862851801202L;
	public static final Point NO_KO = new Point(-2, -2);

	private final int x;
	private final int y;

	public Point(int x, int y) {
		this.x = x;
		this.y = y;
	}

	public int getX() {
		return x;
	}

	public int getY() {
		return y;
	}	

	@Override
	public boolean equals(Object obj) {
		if (obj instanceof Point) {
			Point p = (Point) obj;
			return p.x == x && p.y == y;
		} else {
			return false;
		}
	}
	@Override
	public String toString() {
		return "(" + x + ", " + y + ")";
	}
}

File: app/src/main/java/fr/narwhals/go/domain/Point.kt
package fr.narwhals.go.domain

import java.io.Serializable

data class Point(val x: Int, val y: Int) : Serializable {

    companion object {
        @JvmField val SENTINEL = Point(0, 0);
        @JvmField val PASS  = Point(-1, -1)
        @JvmField val NO_KO = Point(-2, -2)
    }

    override fun toString() = when (this) {
        PASS -> ""
        else -> String(charArrayOf('a'.plus(x - 1), 'a'.plus(y - 1)))
    }

    override fun equals(that: Any?): Boolean {
        if (that is Point) {
            return this.x == that.x && this.y == that.y
        } else {
            return false;
        }
    }

    override fun hashCode(): Int {
        return x - 1 + 52 * (y - 1)
    }
}
